### PR TITLE
Nikita/dev

### PR DESCRIPTION
**Pull request by krigar1184**

### CODE REVIEW

File: Lucidtech/Lucidtech.csproj [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <AssemblyName>Lucidtech</AssemblyName>
     <EnableDefaultCompileItems>False</EnableDefaultCompileItems>
     <FileAlignment>512</FileAlignment>
-    <TargetFrameworks>net461;net47;net472</TargetFrameworks>
+    <TargetFrameworks>netstandard20</TargetFrameworks>
```
Comment (magnurud) [inline Lucidtech/Lucidtech.csproj:15]: I believe we still want to support the existing frameworks.

File: Test/Test.cs [hunk 2]
```diff
@@ -334,7 +330,7 @@ public void TestCreateBatch()
             var transitionId = $"las:transition:{Guid.NewGuid().ToString()}";
             var response = Toby.ListTransitionExecutions(
                 transitionId,
-                new List<string>{ status },
+                status,
```
Comment (magnurud) [inline Test/Test.cs:333]: Status can still be a list of strings, so this should still be tested IMO.

File: Lucidtech/Las/Client.cs [hunk 3]
```diff
@@ -488,22 +666,46 @@ public object CreateBatch(string description)
...
+        /// <param name="executionIds">List of execution ids to filter by</param>
+        /// <param name="maxResults">Maximum number of results to be returned</param>
+        /// <param name="nextToken">A unique token used to retrieve the next page</param>
+        /// <param name="sortBy">The sorting variable of the execution: "endTime" | "startTime"</param>
+        /// <param name="order">Order of the executions: "ascending" | "descending"</param>
+        /// <returns>Transition executions response from the REST API</returns>
         public object ListTransitionExecutions(
             string transitionId,
-            List<string>? statuses = null,
+            string? status = null,
```
Comment (magnurud) [inline Lucidtech/Las/Client.cs:700]: Status can be a list of several statuses or a single status.

File: Lucidtech/Las/Client.cs [hunk 4]
```diff
@@ -527,6 +729,69 @@ public object CreateBatch(string description)
...
+            List<string>? status = null,
+            List<string>? executionIds = null,
+            int? maxResults = null,
+            string? nextToken = null,
+            string? sortBy = null,
+            string? order = null
+        ) {
+            var queryParams = new Dictionary<string, object> {
+                {"status", status},
+                {"executionId", executionIds},
```
Comment (magnurud) [inline Lucidtech/Las/Client.cs:758]: Does this work? will the query-parameters be appended as expected? 
`?executionId=a&executionId=b....`

**Review by magnurud — CHANGES_REQUESTED** [review]

File: Lucidtech/Las/Client.cs [hunk 5]
```diff
@@ -35,6 +35,17 @@ public Client(Credentials credentials)
         /// </summary>
         public Client() : this(new Credentials()) {}
 
+        /// <summary>Creates an asset handle, calls the POST /assets endpoint.</summary>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:38]: Can exclude the word 'handle' here I think

File: Lucidtech/Las/Client.cs [hunk 6]
```diff
@@ -51,6 +62,21 @@ public Client(Credentials credentials)
             return ExecuteRequestResilient(RestSharpClient, request);
         }
 
+        /// <summary> List available assets, calls the GET /assets endpoint. </summary>
+        /// <example>
+        /// Create a document handle for a jpeg image
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:67]: This sounds like documentation from /documents?

File: Lucidtech/Las/Client.cs [hunk 7]
```diff
@@ -195,34 +243,33 @@ public Client(Credentials credentials)
...
+        public object GetDocument(string documentId) {
             RestRequest request = ClientRestRequest(Method.GET, $"/documents/{documentId}");
             return ExecuteRequestResilient(RestSharpClient, request);
         }
 
         /// <summary>
-        /// Post feedback to the REST API, calls the POST /documents/{documentId} endpoint.
-        /// Posting feedback means posting the ground truth data for the particular document.
+        /// Post ground truth to the REST API, calls the POST /documents/{documentId} endpoint.
+        /// Posting ground truth means posting the ground truth data for the particular document.
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:253]: Maybe 'post' is a bit misleading here, should we change it to 'Update ground truth for the document' or something similar?

File: Lucidtech/Las/Client.cs [hunk 8]
```diff
@@ -385,6 +492,19 @@ public object CreateBatch(string description)
             return ExecuteRequestResilient(RestSharpClient, request);
         }
 
+        /// <summary>Updates an secret, calls the PATCH /secrets/secretId endpoint.</summary>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:495]: an secret -> a secret

File: Lucidtech/Las/Client.cs [hunk 9]
```diff
@@ -356,6 +435,19 @@ public object CreateBatch(string description)
             return ExecuteRequestResilient(RestSharpClient, request);
         }
 
+        /// <summary>Creates an secret handle, calls the POST /secrets endpoint.</summary>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:438]: Create a secret?

File: Lucidtech/Las/Client.cs [hunk 10]
```diff
@@ -406,19 +526,53 @@ public object CreateBatch(string description)
...
+        ///     {"title", "output"}
+        /// };
+        /// var params = new Dictionary&lt;string, object&gt;{
+        ///     {"imageUrl", "&lt;image_url&gt;"},
+        ///     {"credentials", new Dictionary&lt;string, string&gt;{
+        ///         {"username", "&lt;username&gt;"},
+        ///         {"password", "&lt;password&gt;"}
+        ///     }
+        /// };
+        /// var response = client.CreateTransition(inputSchema, outputSchema, parameters: params);
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:548]: missing transitionType in example call

File: Lucidtech/Las/Client.cs [hunk 11]
```diff
@@ -406,19 +526,53 @@ public object CreateBatch(string description)
...
+        /// </code>
+        /// </example>
+        /// <param name="transitionType">Type of transition: "docker"|"manual"</param>
+        /// <param name="inputJsonSchema">Json-schema that defines the input to the transition</param>
+        /// <param name="outputJsonSchema">Json-schema that defines the output of the transition</param>
+        /// <param name="parameters">Parameters to the corresponding transition type</param>
+        /// <param name="optionalParams">Additional attributes</param>
+        /// <returns>Transition response from REST API</returns>
         public object CreateTransition(
             string transitionType,
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:558]: Should this be an enum?

File: Lucidtech/Las/Client.cs [hunk 12]
```diff
@@ -488,22 +666,46 @@ public object CreateBatch(string description)
...
+        /// var response = client.ListTransitionExecutions("&lt;transitionId&gt;", new [] {"succeeded", "failed"});
+        /// </code>
+        /// </example>
+        /// <param name="transitionId">Id of the transition</param>
+        /// <param name="status">Status to filter by</param>
+        /// <param name="executionIds">List of execution ids to filter by</param>
+        /// <param name="maxResults">Maximum number of results to be returned</param>
+        /// <param name="nextToken">A unique token used to retrieve the next page</param>
+        /// <param name="sortBy">The sorting variable of the execution: "endTime" | "startTime"</param>
+        /// <param name="order">Order of the executions: "ascending" | "descending"</param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:696]: Should this be an enum?

File: Lucidtech/Las/Client.cs [hunk 13]
```diff
@@ -488,22 +666,46 @@ public object CreateBatch(string description)
...
 
+        /// <summary>List executions in a transition, calls the GET /transitions/{transitionId}/executions endpoint.</summary>
+        /// <example>
+        /// <code>
+        /// Client client = new Client();
+        /// var response = client.ListTransitionExecutions("&lt;transitionId&gt;", new [] {"succeeded", "failed"});
+        /// </code>
+        /// </example>
+        /// <param name="transitionId">Id of the transition</param>
+        /// <param name="status">Status to filter by</param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:691]: Should this be an enum?

File: Lucidtech/Las/Client.cs [hunk 14]
```diff
@@ -527,6 +729,69 @@ public object CreateBatch(string description)
...
 
+        /// <summary>List executions in a transition, calls the GET /transitions/{transitionId}/executions endpoint.</summary>
+        /// <example>
+        /// <code>
+        /// Client client = new Client();
+        /// var response = client.ListTransitionExecutions("&lt;transitionId&gt;", new [] {"succeeded", "failed"});
+        /// </code>
+        /// </example>
+        /// <param name="transitionId">Id of the transition</param>
+        /// <param name="statuses">List of execution statuses to filter by</param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:740]: param called 'statuses' here, but 'status' in actual function

File: Lucidtech/Las/Client.cs [hunk 15]
```diff
@@ -573,16 +865,89 @@ public object CreateBatch(string description)
...
+        /// };
+        /// var response = client.UpdateUser("&lt;user_id&gt;", parameters);
+        /// </code>
+        /// </example>
+        /// <param name="userId">Id of the user</param>
+        /// <param name="attributes">
+        /// Attributes to update.
+        /// Allowed attributes:
+        ///     name (string),
+        ///     avatar (base64-encoded image)
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:911]: Are we a bit inconsisten in how we name optional params to a function?
attributes
optionalParams
etc

File: Lucidtech/Las/Client.cs [hunk 16]
```diff
@@ -639,6 +1048,68 @@ public object CreateBatch(string description)
...
+        /// <param name="workflowId">Id of the workflow</param>
+        /// <param name="statuses">Workflow execution status to filter by</param>
+        /// <param name="maxResults">Maximum number of results to be returned</param>
+        /// <param name="nextToken">A unique token used to retrieve the next page</param>
+        /// <param name="sortBy">The sorting variable of the execution: "endTime" | "startTime"</param>
+        /// <param name="order">Order of the executions: "ascending" | "descending"</param>
+        /// <returns>WorkflowExecutions response from REST API</returns>
+        public object ListWorkflowExecutions(
+            string workflowId,
+            string? status = null,
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:1068]: status or statuses?

File: Lucidtech/Las/Client.cs [hunk 17]
```diff
@@ -639,6 +1048,68 @@ public object CreateBatch(string description)
...
+        /// <summary>List executions in a workflow, calls the GET /workflows/{workflowId}/executions endpoint.</summary>
+        /// <example>
+        /// <code>
+        /// Client client = new Client();
+        /// var statuses = new [] {"running", "succeeded"};
+        /// var response = client.ListWorkflowExecutions("&lt;workflow_id&gt;", statuses);
+        /// </code>
+        /// </example>
+        /// <param name="workflowId">Id of the workflow</param>
+        /// <param name="statuses">Workflow execution status to filter by</param>
```
Comment (augustak) [inline Lucidtech/Las/Client.cs:1060]: status or statuses?

File: Lucidtech/Las/Client.cs [hunk 18]
```diff
@@ -746,17 +750,22 @@ public object CreateBatch(string? name = null, string? description = null)
         /// <returns>Transition executions response from the REST API</returns>
         public object ListTransitionExecutions(
             string transitionId,
-            List<string>? status = null,
+            List<string>? statuses = null,
```
Comment (magnurud) [inline Lucidtech/Las/Client.cs]: Wouldn't it be less code to maintain if the previous method calls this method with its status as a list with one element?

File: Lucidtech/Las/Client.cs [hunk 19]
```diff
@@ -1057,7 +1066,7 @@ public object CreateBatch(string? name = null, string? description = null)
         /// </code>
         /// </example>
         /// <param name="workflowId">Id of the workflow</param>
-        /// <param name="statuses">Workflow execution status to filter by</param>
+        /// <param name="status">Workflow execution status to filter by</param>
```
Comment (magnurud) [inline Lucidtech/Las/Client.cs:1069]: This method could also call the other one.

**Review by magnurud — CHANGES_REQUESTED** [review]

File: Test/Test.cs [hunk 20]
```diff
@@ -143,6 +142,7 @@ public void TestCreatePredictionMaxPages()
             CheckKeys(expectedKeys, response);
         }
 
+        [Ignore("")]
```
Comment (magnurud) [inline Test/Test.cs:145]: Could you add a short description to why this is ignored?

**Review by magnurud — CHANGES_REQUESTED** [review]

**Review by magnurud — APPROVED** [review]

**Pull request merged**